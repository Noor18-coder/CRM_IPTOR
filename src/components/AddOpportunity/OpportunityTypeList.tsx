import React from "react";
import { OpportunityType } from "../../helpers/Api/models";
import VectorImg from "../../assets/images/check_circle.svg";
import { Image} from "react-bootstrap";

export interface Props {
  opptyTypes : OpportunityType[];
  doClick: (key: string) => void;
  selected?:string
}

export const OpportunityTypeList: React.FC<Props> = ({ opptyTypes, doClick, selected }) => {
  return (
    <ul className="opptytype-list-item">
      {opptyTypes.map((obj:OpportunityType) => {
        return (
          <li key={obj.description}  onClick={() => doClick(obj.description)}>
            <div
              className={"company-container"}
              key={obj.description}
            >
              <div className={"center"}>
                <div className={"test"}>{obj.oppRecordType}</div>
                <Image
                  className={"company-selection-img"}
                  style={{ display: obj.description  === selected ? "block" : "none" }}
                  src={VectorImg}
                  alt="company"
                  title="company"
                ></Image>
              </div>
            </div>
          </li>
        );
      })}
    </ul>
  );
};
