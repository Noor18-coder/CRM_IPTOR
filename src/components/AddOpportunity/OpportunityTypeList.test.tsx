import renderer from 'react-test-renderer';
import { OpportunityTypeList } from './OpportunityTypeList';
import OpportunityTypeMock from '../../mocks/OpportunityType.mock';

const opptyTypes = OpportunityTypeMock.getOpportunityTypesgetConstantValue();
const doClick = jest.fn();

describe('[AddOpportunity] OpportunityTypeList', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<OpportunityTypeList opptyTypes={opptyTypes} doClick={doClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
