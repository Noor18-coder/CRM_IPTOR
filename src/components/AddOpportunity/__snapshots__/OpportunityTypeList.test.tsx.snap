// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[AddOpportunity] OpportunityTypeList should renders correctly 1`] = `
<ul
  className="opptytype-list-item"
>
  <li
    onClick={[Function]}
    role="presentation"
  >
    <div
      className="company-container"
    >
      <div
        className="center"
      >
        <div
          className="test"
        >
          Avon
        </div>
        <img
          alt="company"
          className="company-selection-img"
          src="check_circle.svg"
          style={
            Object {
              "display": "none",
            }
          }
          title="company"
        />
      </div>
    </div>
  </li>
</ul>
`;
